resources/default.rb

```diff
@@ -0,0 +1,139 @@
+#
+# Cookbook:: golang
+# Resource:: default
+#
+# Copyright:: 2020, Sous-chefs
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not
+# use this file except in compliance with the License. You may obtain a copy
+# of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+#
+
+# Install golang by compiling from source
+property :from_source, [true, false], default: false
+
+# Base directory for go
+property :install_dir, String, default: '/usr/local'
+property :gopath, String, default: '/opt/go'
+property :gobin, String, default: '/opt/go/bin'
+
+# Directory and file attributes
+property :owner, String, default: 'root'
+property :group, String, default: 'root'
+property :directory_mode, String, default: '0755'
+
+# version to install
+property :version, String, default: '1.14.4'
+property :source_version, String, default: '1.14.4'
+# ark input source
+property :source_url, String, required: true
+# ark input binaries and stuff download
+property :url, String, required: true
```

Comment:
@MarkGibbons one suggestion here: like the other properties, I think these should have default URLs, which mimic the attributes:

```ruby
default['golang']['url'] = "https://dl.google.com/go/#{node['golang']['filename']}"

default['golang']['source_url'] = "https://dl.google.com/go/#{node['golang']['source_filename']}"
```

That will ease the _next_ version, where we drop the attributes & recipes entirely.

Reply:
Thanks, makes sense. I'll see what I can do.